stan/math/prim/meta/macros.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef STAN_MATH_PRIM_META_MACROS_HPP
+#define STAN_MATH_PRIM_META_MACROS_HPP
```

Comment:
Macros is a bit too generic name for this file. How about `compiler_attributes` or `compiler_specific`?

Reply:
I like `compiler_attributes`